.github/workflows/release-move-tracking-tag.yml

```diff
@@ -0,0 +1,32 @@
+name: Release - Move Tracking Tag
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  Move-Tracking-Tag-To-Latest-Release:
+    runs-on: ubuntu-latest
+
+    # TODO: Should we filter this job...? Ie, what if the new release isn't a
+    # version? Or if it's an older patch version rather than a newer one?
+    # The logic gets tricky quickly... so IMO either we assume every new release
+    # is newest and always bump on release publish, or we switch to letting letting the
+    # release author decide by switching this to using `workflow_dispatch` and
+    # requiring a human to manually run this workflow after they tag a release.
+    # Safer, but more of a hassle and also an unnecessary precaution 99% of the time.
```

Comment:
I could use feedback here... do we go for convenience and bump tag on every release, or safety and require manually triggering this workflow after a release is tagged?

i'd probably go with simplicity for now given how infrequently we release in this repo, we are unlikely to backport fixes to a prior release.